src/main/kotlin/no/brreg/informasjonsforvaltning/organizationcatalogue/controller/OrganizationsApiImpl.kt

```diff
@@ -69,6 +69,11 @@ open class OrganizationsApiImpl(
         }
     }
 
+    override fun getDelegatedOrganizations(httpServletRequest: HttpServletRequest?): ResponseEntity<List<Organization>> {
```

Comment:
Jeg tenkte egentlig at det var like så greit å utvide 'getOrganizations', men ja, greit nok.

Reply:
problemet med å utvide den er at hvis jeg spør en url som /organizations?name=EttEllerAnnetAS&allowedDelegatedRegistration=true

Da forventer jeg at jeg får en liste av selskap (ett selskap i vårt tilfelle) med de query parametrene. Men det jeg ender opp med å få er en liste som inneholder ett selskap (selvom allowedDelegatedRegistration er kanskje false" pga switch (when) statement. Det burde ikke ha vært noe switch statement, men heller en findAll() metode som filtrerer mot alle query params som ikke er null.

Reply:
Jepp, as is fungerer ikke getOrganizations optimalt. Så da kunne vi brukt dette som en unnskyldning til å i tillegg fikse det endepunktet ;)

Reply:
Det kan vi :) Men tar det etter test issuen i en separat PR

Reply:
Jeg ser ikke noe poeng å legge mer "if-setninger" til samme endepunkt. Semantisk er organisasjonene som kan registrere på vegne av andre et subset av organisasjoner og er dermed exponert under tilsvarende uri.